sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,280 @@
+import { saveMSFAATestInputsData } from "./msfaa-factory";
+import { In, IsNull } from "typeorm";
+
+describe(
+  describeProcessorRootTest(QueueNames.FullTimeMSFAAProcessResponseIntegration),
+  () => {
+    let app: INestApplication;
+    let processor: FullTimeMSFAAProcessResponseIntegrationScheduler;
+    let db: E2EDataSources;
+    let sftpClientMock: DeepMocked<Client>;
+    let msfaaMocksDownloadFolder: string;
+
+    beforeAll(async () => {
+      msfaaMocksDownloadFolder = path.join(__dirname, "msfaa-receive-files");
+      // Set the ESDC response folder to the files mocks folders.
+      process.env.ESDC_RESPONSE_FOLDER = msfaaMocksDownloadFolder;
+      const { nestApplication, dataSource, sshClientMock } =
+        await createTestingAppModule();
+      app = nestApplication;
+      db = createE2EDataSources(dataSource);
+      sftpClientMock = sshClientMock;
+      // Processor under test.
+      processor = app.get(FullTimeMSFAAProcessResponseIntegrationScheduler);
+    });
+
+    beforeEach(async () => {
+      jest.clearAllMocks();
+      // Force any not signed MSFAA to be signed to ensure that new
+      // ones created will be the only ones available to be updated.
+      await db.msfaaNumber.update(
+        { dateSigned: IsNull() },
+        { dateSigned: getISODateOnlyString(new Date()) },
+      );
+      // Cancel any not canceled MSFAA.
+      await db.msfaaNumber.update(
+        { cancelledDate: IsNull() },
+        { cancelledDate: getISODateOnlyString(new Date()) },
+      );
+    });
+
+    it("Should reactivate a cancelled MSFAA when the same MSFAA is received in the response file.", async () => {
+      // Arrange
+      const student = await saveFakeStudent(db.dataSource);
+      const msfaaInputData = [MSFAA_FULL_TIME_MARRIED];
+      const createdMSFAARecords = await saveMSFAATestInputsData(
+        db,
+        msfaaInputData,
+        { student },
+        { state: MSFAAStates.CancelledOtherProvince },
+      );
+      // Queued job.
+      const job = createMock<Job<void>>();
+      mockDownloadFiles(sftpClientMock, [
+        MSFAA_FULL_TIME_RECEIVE_FILE_WITH_REACTIVATION_RECORD,
+      ]);
+      // Act
+      const processResult = await processor.processMSFAA(job);
+      // Assert
+      expect(processResult).toStrictEqual([
+        {
+          processSummary: [
+            `Processing file ${MSFAA_FULL_TIME_RECEIVE_FILE_WITH_REACTIVATION_RECORD}.`,
+            "File contains:",
+            "Confirmed MSFAA records (type R): 1.",
+            "Cancelled MSFAA records (type C): 0.",
+            "Record from line 1, updated as confirmed.",
+          ],
+          errorsSummary: [],
+        },
+      ]);
+      // Assert that the file was deleted from SFTP.
+      expect(sftpClientMock.delete).toHaveBeenCalled();
+      // Find the updated MSFAA records previously created.
+      const msfaaIDs = createdMSFAARecords.map((msfaa) => msfaa.id);
```

Comment:
Only one MSFAA record is created for this test, so we do not need this type of logic, IMO.